File: Python/Easy/Is-Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = 0
        if len(s) is 0:
            return True
        for char in t:
            if i < len(s):
                if s[i] is char:
                    i+=1
            else:
                return i==len(s)
        return i==len(s)
        

File: Java/Easy/Is-Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int n = t.length();
        int sL = s.length();
        int counter = 0;
        for (int i = 0; i<n; i++) {
            if (counter<sL) {
                if (s.charAt(counter)==t.charAt(i)) {
                    counter++;
                }
            } else {
                return sL==counter;
            }
        }
        return sL==counter;
    }
}
